python_modules/dagster/dagster/_core/workspace/load_target.py

```diff
@@ -41,21 +41,44 @@ def create_origins(self) -> Sequence[CodeLocationOrigin]:
         return location_origins_from_yaml_paths(self.paths)
 
 
+def is_valid_modules_list(modules: Any) -> bool:
+    if isinstance(modules, list):
+        for item in modules:
+            if not (isinstance(item, dict) and 'type' in item and 'name' in item):
+                return False
+        return True
+    return False
+
+
 def get_origins_from_toml(path: str) -> Sequence[ManagedGrpcPythonEnvCodeLocationOrigin]:
     with open(path, "rb") as f:
         data = tomli.load(f)
         if not isinstance(data, dict):
             return []
 
         dagster_block = data.get("tool", {}).get("dagster", {})
+        origins = []
+
         if "module_name" in dagster_block:
-            return ModuleTarget(
+            origins += ModuleTarget(
                 module_name=dagster_block["module_name"],
                 attribute=None,
                 working_directory=os.getcwd(),
                 location_name=dagster_block.get("code_location_name"),
             ).create_origins()
-        return []
+
+        if "modules" in dagster_block and is_valid_modules_list(dagster_block["modules"]):
+            for module in dagster_block["modules"]:
+                if module["type"] == "module":
+                    origins += ModuleTarget(
+                        module_name=module["name"],
+                        attribute=None,
+                        working_directory=os.getcwd(),
+                        location_name=dagster_block.get("code_location_name"),
+                    ).create_origins()
+
+        return origins
+
```

Comment:
I think we should throw an error if a user specific both `module` and `modules` in the toml file. This is likely an error and explicitness makes more sense here IMO.

Reply:
@schrockn I agree, I added it and extended the tests to check on that too